### PR TITLE
Add language server CI buildspec; package into bin folder

### PR DESCRIPTION
**Pull request by awschristou**
This change defines a buildspec file to be used as the Language Server CI. I also updated the package script to produce the binaries in a `bin` subfolder, so that the files are easier to identify as produced artifacts.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by JadenSimon — APPROVED** [review]

File: buildspec/languageServerTests.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+version: 0.2
```
Comment (justinmk3) [inline buildspec/languageServerTests.yml:1]: Other files use "lsp" in the name, consider naming this file with the same pattern (`lspTests.yml`).

File: .github/workflows/lsp-ci.yaml [hunk 2]
```diff
@@ -48,5 +48,5 @@ jobs:
             - name: Attach binaries
               uses: actions/upload-artifact@v3
               with:
-                name: langauge-server
-                path: lsp/awsdocuments-language-server-*
+                  name: langauge-server
+                  path: lsp/bin/awsdocuments-language-server-*
```
Comment (justinmk3) [inline .github/workflows/lsp-ci.yaml:52]: I downloaded the artifacts and I see:

```
awsdocuments-language-server-linux
awsdocuments-language-server-macos
awsdocuments-language-server-win.exe
```

That is very verbose. The files could be named `aws-lsp-xx` instead without any ambiguity.

File: buildspec/languageServerTests.yml (hunk 1 above)
Comment (awschristou) [reply buildspec/languageServerTests.yml:1]: lsp does not correctly describe the system, which is a language server that happens to handle the lsp protocol.

File: .github/workflows/lsp-ci.yaml (hunk 2 above)
Comment (awschristou) [reply .github/workflows/lsp-ci.yaml:52]: This is the default behavior of the packager system. We will likely revisit the file names when we get closer to making final distribution bundles.

Comment (awschristou) [reply .github/workflows/lsp-ci.yaml:52]: For example, and TBD: we may have to differentiate x64 vs ARM binaries.

File: buildspec/languageServerTests.yml (hunk 1 above)
Comment (justinmk3) [reply buildspec/languageServerTests.yml:1]: LSP is the common moniker. Is there a risk of confusion? Is there a reasonable chance that any non-LSP lang server will be implemented?

**Pull request merged**